src/test/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensorSpec.scala

```diff
@@ -91,12 +93,36 @@ class ScapegoatSensorSpec extends FlatSpec with SensorContextMatchers with LoneE
 
   it should "not get a module file if its filename does not match any file" in {
     val filesystem = new DefaultFileSystem(Paths.get("./"))
+
     val moduleFile =
       scapegoatSensor.getModuleFile("com/mwz/sonar/scala/scapegoat/TestFileA.scala", filesystem)
 
     moduleFile shouldBe None
   }
 
+  it should "not get a module file if its filename match two or more files" in {
```

Comment:
This is the only thing that bothers me about the current implementation...

But, I think this is not common. What do you think?

Reply:
I'm not sure if I understand what you're testing here since in case of multi-module projects, the file system only contains files from one module.

Reply:
Probably I should have used something different than _"modules"_.

Well, the problem is the same than with Scoverage.

The filenames in the Scapegoat report are relative to the package folder. Thus source prefixes like _"src/main/scala"_ aren't present.

This time I decided that the best thing to do was to ask the sonarqube filesystem for files whose path ends with the filename from the Scapegoat report.

But if two (or more files) match, that would be an error.

Reply:
I think I understand it now. So in this test you set up a filesystem in a current directory and you define two files with the same name being located in different "modules" and you try to get an `InputFile` from the filesystem for a file path which comes from a scapegoat report.

I don't think this scenario is possible to happen in this configuration for the following reasons:
 - if this was a single module project, the prefix would be `src/main/scala` and you would expect both `moduleA` and `moduleB` to be a part of the full package name of the files, so they would end up being unique in the single scapegoat report.

 - for a multi-module project, the sensor would be run separately for each module, so you would end up processing two independent scapegoat reports - end even though each would contain a file with the same name and package name, e.g. `com.mwz.sonar.scala.scapegoat.TestFile.scala`, the filesystem would only give you one file from the current module which is being analysed.

Reply:
Excellent, I see now what you meant - thanks for calryfiying this 👍 

> The problem here is if there are two files with the same package and filename but two different prefixes. e.g. "src/main/scala" & "src/main/java".
> 
> But I believe that is also not common (maybe even not valid for the compiler, I'm not sure).
> So I think we both agree it is good enough. Or what do you think?

Yeah, I also don't think it's valid and from a practical point of view I don't see any value in having Scala files mixed with Java in the same sources directory, so we can probably just ignore this case.

> Also, I think this is ready to be merged, unless you have any other suggestion. (like removing this specific test maybe?).

Great - it looks good to me. Let's remove this test then and I'll get it merged.